src/lib.rs

```diff
@@ -204,6 +364,15 @@ mod tests {
         assert_eq!(adler32_slice(&[0xff; 1024 * 1024]), 0x8e88ef11);
     }
 
+    #[test]
+    fn unaligned() {
+        let buf = [0xe7; 1024];
+        assert_eq!(adler32_slice(&buf[0..1021]), 0x6c739979);
+        assert_eq!(adler32_slice(&buf[1..1022]), 0x6c739979);
+        assert_eq!(adler32_slice(&buf[2..1023]), 0x6c739979);
+        assert_eq!(adler32_slice(&buf[3..]), 0x6c739979);
```

Comment:
Just to make it slightly easier to read:
```suggestion
        assert_eq!(adler32_slice(&buf[3..1024]), 0x6c739979);
```